beluga_amcl/include/beluga_amcl/ndt_amcl_node_3d.hpp

```diff
@@ -145,6 +145,9 @@ class NdtAmclNode3D : public BaseAMCLNode {
   std::optional<Sophus::SE3d> last_known_odom_transform_in_map_;
   /// Whether to broadcast transforms or not.
   bool enable_tf_broadcast_{false};
+
+  /// Map representation publisher.
+  rclcpp::Publisher<visualization_msgs::msg::MarkerArray>::SharedPtr map_visualization_pub_;
```

Comment:
@Fernando-Sanz meta: we should be using `rclcpp::LifecyclePublisher`s for this

Reply:
IMHO it does not make any real difference.
A lifecycle publisher only provides an on/off, that will discard the message if you publish before activating.
It's not really useful.

As long as you create the publisher on configure and start publishing after activation, then it's fine.

Reply:
Still, consistency with the rest of the codebase is good.